docs/source/mtraining.rst

```diff
@@ -4,6 +4,190 @@
+- ``List<Course> getAllCourses()``, ``List<Chapter> getAllChapters()``, ``List<Lesson> getAllLessons()``, ``List<Quiz> getAllQuizzes()`` - returns all unused units.
+- ``Quiz getQuizForChapter(long id)`` - returns the quiz for a chapter with the given id.
+
+Bookmark Service
+################
+
+Service for management of course bookmarks for a user. This is used to maintain the progress of a user in the curriculum.
+
+- ``Bookmark createBookmark(Bookmark bookmark)`` - creates a bookmark for a user
+- ``Bookmark getBookmarkById(long id)`` - returns a bookmark for the given id.
+- ``Bookmark getLatestBookmarkByUserId(String externalId)`` - returns the latest bookmark for the user identified by the externalId.
+- ``List<Bookmark> getAllBookmarksForUser(String externalId)`` - returns all bookmarks for the user identified by the externalId.
+- ``Bookmark updateBookmark(Bookmark bookmark)`` - updates the given bookmark.
+- ``void deleteBookmark(long id)`` - deletes a bookmark with the given id.
+- ``void deleteAllBookmarksForUser(String externalId)`` - deletes all bookmarks for the user identified by the externalId.
+
+Activity Record Service
+#######################
+
+Service for managing activity records.
+
+- ``ActivityRecord createActivity(ActivityRecord activityRecord)`` - creates the given activity record.
+- ``ActivityRecord updateActivity(ActivityRecord activityRecord)`` - updates the given activity record.
+- ``ActivityRecord getActivityById(long activityId)`` - returns an activity record by the given id.
+- ``List<ActivityRecord> getAllActivityForUser(String externalId)`` - returns all activity records for a user with the given externalId.
+- ``List<ActivityRecord> getCompletedActivityForUser(String externalId)`` -  returns all completed activity records for user identified by the externalId.
+- ``void deleteActivity(long activityRecordId)`` - deletes the activity record by the given id.
+- ``void deleteAllActivityForUser(String externalId)`` - deletes all activity records for a user with the given externalId.
+
+############
+mTraining UI
+############
+
+The mTraining module has ``Courses``, ``Chapters``, ``Lessons`` and ``Quizzes`` tabs which allows to perform CRUD operations
+on the course units. ``Bookmarks`` and ``Activity Record`` tabs allows to view logs. All those tab are using the MDS embedded UI.
+
+Default tab is the ``Tree View`` tab. ``TreeView`` tab allows to manage relationships between course units. Colors of the units
+represents state(``blue`` - ``Active``, ``grey`` - ``Inactive``, ``orange`` - ``Pending``). You can easily change state of
+the entire branch or a single unit, to do this use buttons below the tree. When you will use ``Remove member`` button then
+all children of the element will be disconnected(for example when you remove chapter then all ``Lessons`` will be disconnected form
+``Chapter`` (each ``Lesson`` will be unused)). To add unit to the tree you must select node and then drag unit from ``Nodes``
```

Comment:
Mention that you mean the node of the tree. Also add that UI will display all still available units to add.